pkg/clustermesh/kvstoremesh/kvstoremesh.go

```diff
@@ -80,6 +86,31 @@ func newKVStoreMesh(lc cell.Lifecycle, params params) *KVStoreMesh {
 	return &km
 }
 
+type SyncWaiterParams struct {
+	cell.In
+
+	KVStoreMesh *KVStoreMesh
+	SyncState   health.SyncState
+}
+
+func RegisterSyncWaiter(lc cell.Lifecycle, p SyncWaiterParams) {
+	lc.Append(cell.Hook{
+		OnStart: func(ctx cell.HookContext) error {
+			syncCtx, cancel := context.WithTimeout(ctx, p.KVStoreMesh.config.GlobalReadyTimeout)
```

Comment:
I'd move the timeout to the synced function, given that the parameter is part of kvstoremesh nonetheless.